server.go

```diff
@@ -803,33 +804,75 @@ func newServer(cfg *Config, listenAddrs []net.Addr,
 	// replicate it, and instead it'll only be stored locally.
 	chanGraph := dbs.GraphDB.ChannelGraph()
 
+	// Fetch the source node from the channel graph. If a source node is
+	// found, we'll use its values for our initial node announcement.
+	persistedNode, err := chanGraph.SourceNode()
+	if err != nil {
+		srvrLog.Errorf("unable to fetch source node: %v", err)
+	}
+
 	// We'll now reconstruct a node announcement based on our current
 	// configuration so we can send it out as a sort of heart beat within
 	// the network.
 	//
-	// We'll start by parsing the node color from configuration.
-	color, err := lncfg.ParseHexColor(cfg.Color)
-	if err != nil {
-		srvrLog.Errorf("unable to parse color: %v\n", err)
-		return nil, err
-	}
 
-	// If no alias is provided, default to first 10 characters of public
-	// key.
-	alias := cfg.Alias
-	if alias == "" {
-		alias = hex.EncodeToString(serializedPubKey[:10])
+	// If the node is restarted, we should use the previously persisted
+	// values for the node's alias, color, addresses, and features.
+	// Otherwise, we'll use the configured values.
+	var (
+		color    color.RGBA
+		alias    string = cfg.Alias
+		features *lnwire.FeatureVector
+	)
+
+	// If the node was previously persisted, we'll use the saved values.
+	if persistedNode != nil {
```

Comment:
haven't checked the details, but I think we should create a method instead of growing the already giant `newServer` method.

Reply:
I agree. I've refactored the code into a method